### PR TITLE
functionality for legacy messages rendering in holiday modal

### PR DESCRIPTION
**Pull request by PaulTummonUnosquare**

### CODE REVIEW

File: web-app/src/components/BookingModal/container.js [hunk 1]
```diff
@@ -10,13 +10,24 @@ import {
...
+{ author: 'Spider Man', eventMessageId: 6, lastModified: '2018-09-19', message: 'But Why???', messageTypeDescription: 'General', messageTypeId: 1 },
+{ author: 'Spider Man', eventMessageId: 6, lastModified: '2018-09-19', message: 'But Why???', messageTypeDescription: 'General', messageTypeId: 1 },
+{ author: 'Spider Man', eventMessageId: 6, lastModified: '2018-09-19', message: 'But Why???', messageTypeDescription: 'General', messageTypeId: 1 },
+{ author: 'Paul Tummon', eventMessageId: 1, lastModified: '2018-09-19', message: 'Sorry Spiderman, No go dsjfdfskgfdskjfhdkjshfsdkfhskdfhdask', messageTypeDescription: 'General', messageTypeId: 1 },
+{ author: 'Paul Tummon', eventMessageId: 1, lastModified: '2018-09-19', message: 'Sorry Spiderman, No go', messageTypeDescription: 'General', messageTypeId: 1 },
+{ author: 'Paul Tummon', eventMessageId: 1, lastModified: '2018-09-19', message: 'Sorry Spiderman, No go', messageTypeDescription: 'General', messageTypeId: 1 },
+{ author: 'Paul Tummon', eventMessageId: 1, lastModified: '2018-09-19', message: 'Sorry Spiderman, No go', messageTypeDescription: 'General', messageTypeId: 1 }];
+
```
Comment (PaulTummonUnosquare) [inline web-app/src/components/BookingModal/container.js]: This is hard coded test data as the endpoint isn't as yet feeding back the legacy messages.

File: web-app/src/components/BookingModal/container.js [hunk 2]
```diff
@@ -163,15 +182,39 @@ const Container = Wrapped =>
         });
     };
 
+    getLegacyMessages = () => {
+      const { legacyMessages } = this.state;
+      event.preventDefault();
```
Comment (William-Unosquare) [inline web-app/src/components/BookingModal/container.js]: where's `event` coming from here?

File: web-app/src/components/BookingModal/container.js [hunk 3]
```diff
@@ -163,15 +182,39 @@ const Container = Wrapped =>
...
+            legacyMessages={legacyMessages}
+            legacyHolidayMessagelist={this.legacyHolidayMessagelist}
             submitRejectionResponse={this.submitRejectionResponse}
+            toggleRejectionMessageView={toggleRejectionMessageView}
             rejectionResponseText={this.state.rejectionResponseText}
             assignRejectionResponseText={this.assignRejectionResponseText}
             booking={this.props.booking}
-            toggleRejectionResponseView={this.state.toggleRejectionResponseView}
+            toggleLegacyHolidayMessageView={this.toggleLegacyHolidayMessageView}
+            toggleRejectionResponseView={toggleRejectionResponseView}
```
Comment (William-Unosquare) [inline web-app/src/components/BookingModal/container.js:184]: Some of this stuff definitely needs to be extracted into its own component in the future. The amount of props we're passing down is getting a bit mad.

File: web-app/src/components/BookingModal/index.js [hunk 4]
```diff
@@ -54,31 +86,33 @@ const BookingModal = props => {
             toggleRejectionMessageInputView={toggleRejectionMessageInputView}
           />
         )}
-        <h1>
-          {isEventBeingUpdated ? 'Update Booking' : 'Request a Holiday'}
+        {toggleRejectionMessageView && (<div><h1>Message History</h1>{legacyHolidayMessagelist(legacyMessages)}</div>)}
+        {!toggleRejectionMessageView && (<div><h1>
```
Comment (William-Unosquare) [inline web-app/src/components/BookingModal/index.js]: Pull what you're rendering in these out into their own functions so there's no so many elements relying on an in-line conditional check

File: web-app/src/components/common_styled/Ul.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import styled from 'styled-components';
```
Comment (William-Unosquare) [inline web-app/src/components/common_styled/Ul.js]: This whole file shouldn't exist I think. You've got it in common_styled but there are very specific classes in here, which takes out of "common" territory.

**Review by William-Unosquare — DISMISSED** [review]
The `<Ul>` component needs another looking at. It isn't a common component if you're assigning classes to that are only relevant in one scenario. A couple of other bits of cleanup to look at too but great work again.

File: web-app/src/components/BookingModal/index.js [hunk 6]
```diff
@@ -54,31 +113,18 @@ const BookingModal = props => {
...
-            updateTakenEvents={updateTakenEvents}
-            employeeId={employeeId}
-            booking={booking}
-            bookingDuration={bookingDuration}
-            createEvent={createEvent}
-            updateEvent={updateEvent}
-          />
-        </FormContainer>
+        {renderLegacyMesasge()}
+        {renderBookingModalForm()}
```
Comment (bgribz) [inline web-app/src/components/BookingModal/index.js:95]: These should really be split out into their own components. There's an awful lot going on in these functions that really would be better suited to components.

**Review by bgribz — DISMISSED** [review]

File: web-app/src/components/BookingModal/LegacyMessageList/container.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import Swal from 'sweetalert2';
+import { getAllHolidays } from '../../../services/holidayService';
+
+const testMessageArray = [{ author: 'John Mallon', eventMessageId: 1, lastModified: '2018-09-19', message: 'Sorry, low team capacity', messageTypeDescription: 'General', messageTypeId: 1 }
+  , { author: 'Carl Stevenson', eventMessageId: 6, lastModified: '2018-09-19', message: 'Is there nothing you can do?', messageTypeDescription: 'General', messageTypeId: 1 },
```
Comment (ghost) [inline web-app/src/components/BookingModal/LegacyMessageList/container.js]: Are wanting to merge this into develop?

Comment (PaulTummonUnosquare) [reply web-app/src/components/BookingModal/LegacyMessageList/container.js]: With the demo for the Mexico office team this week, having spoken to William it was agreed this data would be available if the endpoint wasn't ready in time as there have been hard coded elements in the past in development until the endpoints were ready

File: web-app/src/components/BookingModal/LegacyMessageList/styled.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        margin: 7px 10px 13px 7px;
+        color: white;
+        font-family: Open Sans,sans-serif;
+        padding-right: 20px;
+        padding-left: 19px;
+    }
+    
+    height: 300px; 
+    overflow: auto;
+    border: 2px solid #d6d6d6;
```
Comment (ghost) [inline web-app/src/components/BookingModal/LegacyMessageList/styled.js]: Can we use a theme variable here?

File: web-app/src/components/BookingModal/LegacyMessageList/styled.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    text-align: center;
+    
+    .employee-message-2{
+        background-color:#0ecaea;
```
Comment (ghost) [inline web-app/src/components/BookingModal/LegacyMessageList/styled.js]: Just a space here between the hex value and the rule

File: web-app/src/components/BookingModal/index.js [hunk 10]
```diff
@@ -40,11 +43,42 @@ const BookingModal = props => {
     );
   };
 
+  const renderLegacyMesasge = () => {
```
Comment (ghost) [inline web-app/src/components/BookingModal/index.js]: Message is spelt wrong here

File: web-app/src/components/BookingModal/index.js [hunk 11]
```diff
@@ -40,11 +43,42 @@ const BookingModal = props => {
...
+    if (toggleRejectionMessageView) {
+      return <LegacyMessageList />;
+    }
+    return null;
+  };
+
+  const renderBookingModalForm = () => {
+    if (!toggleRejectionMessageView) {
+      return (<div><h1>
```
Comment (ghost) [inline web-app/src/components/BookingModal/index.js]: Could we have the div on a new line etc.

File: web-app/src/components/BookingModal/LegacyMessageList/styled.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
+import styled from 'styled-components';
+
+export const Ul = styled.ul`
+    li {
+        list-style-type: none;
+        margin: 10px 10px 10px 10px;
```
Comment (ghost) [inline web-app/src/components/BookingModal/LegacyMessageList/styled.js]: `margin: 10px;` can be done here

File: web-app/src/components/BookingModal/LegacyMessageList/styled.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+        border-radius: 25px;
+        width: 75%;
+        padding-left: 30px;
+        padding-top: 1px;
+        padding-bottom: 1px;
+        word-wrap:break-word;
+    }
+
+    h1 {
+        margin: 10px 10px 10px 10px;
```
Comment (ghost) [inline web-app/src/components/BookingModal/LegacyMessageList/styled.js]: same as above

File: web-app/src/components/BookingModal/LegacyMessageList/styled.js [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+        color:white;
+    }
+
+    h3 {
+        text-transform: none;
+        margin: 7px 10px 13px 7px;
+        color: white;
+        font-family: Open Sans,sans-serif;
+        padding-right: 20px;
+        padding-left: 19px;
```
Comment (ghost) [inline web-app/src/components/BookingModal/LegacyMessageList/styled.js]: Quite pedantic here but it's been drilled into me, how come one is 20px and one is 19px?

File: web-app/src/components/BookingModal/LegacyMessageList/styled.js (hunk 9 above)
Comment (ghost) [reply web-app/src/components/BookingModal/LegacyMessageList/styled.js]: Plus can we use theme props?

File: web-app/src/components/BookingModal/LegacyMessageList/styled.js [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+    height: 300px; 
+    overflow: auto;
+    border: 2px solid #d6d6d6;
+    text-align: center;
+    
+    .employee-message-2{
+        background-color:#0ecaea;
+    }
+    .employee-message-1{
+        background-color: #0eb5d1;
```
Comment (ghost) [inline web-app/src/components/BookingModal/LegacyMessageList/styled.js]: Theme props?

**Review by ghost — CHANGES_REQUESTED** [review]
A few nit picks there, nicely done though.

File: web-app/src/components/BookingModal/LegacyMessageList/container.js (hunk 7 above)
Comment (ghost) [reply web-app/src/components/BookingModal/LegacyMessageList/container.js]: Perfect, sounds good!

File: web-app/src/components/BookingModal/LegacyMessageList/styled.js (hunk 15 above)
Comment (PaulTummonUnosquare) [reply web-app/src/components/BookingModal/LegacyMessageList/styled.js]: I was following the current format of the project where each distinct component relies on a separate relative style sheet. I think theme props are a good idea but for parity we would need to refactor all the existing style sheet dependencies to bring those styles into the components which might be a separate story in itself?

**Review by ghost — APPROVED** [review]

**Pull request merged**